documentation/src/main/asciidoc/topics/upgrading.adoc

```diff
@@ -18,7 +18,18 @@ The `file-store` element will now install a SoftIndexFileStore instead of the pr
 implementation. Any configuration with a schema of 12 or older will automatically migrate the old
 data storage to the new automatically.
 
-= Upgrading from 12.0 to 12.1
+= Upgrading from 11.x or 12.0 to 12.1
+
+  If you are upgrading from 11.x to 12.1, you should also review the 12.0 to 12.1 upgrade notes.
+
+== Upgrading Deployments with Single File Cache Store Persistence
+
+When upgrading Infinispan from 11.x to 12.1, caches that include a `SingleFileStore` persistence configuration can encounter an issue that leads to data corruption.
+This issue affects Infinispan 12.1.0.Final to 12.1.4.Final.
+As of Infinispan 12.1.5.Final this issue is resolved and data corruption for Single File cache stores no longer occurs.
+
+If you have already upgraded from 11.x to 12.1, you should upgrade to Infinispan 12.1.5.Final or later.
+After completing the upgrade, Infinispan attempts to recover any corrupted data and restore the Single File Store on first start.
```

Comment:
It could be worth adding a highlighted section saying something like: "Before upgrading, we recommend making a backup  of the SFS .dat files in case recovery is not possible"

Reply:
We also need to mention that if the user has any xsite caches that use SFS, it's also recommended for Xsite StateTransfer to be performed once the server(s) have started.